sdk/cosmosdb/cosmos/test/functional/container.spec.ts

```diff
@@ -451,3 +451,51 @@ describe("createIfNotExists", function() {
     assert.equal(def.id, readDef.id);
   });
 });
+
+describe("container.readOffer", function() {
+  let containerWithOffer: Container;
+  let containerWithoutOffer: Container;
+  let container2WithOffer: Container;
+  let container2WithoutOffer: Container;
+  const containerRequestWithOffer: ContainerRequest = {
+    id: "sample",
+    throughput: 400
+  };
+  const containerRequest: ContainerRequest = {
+    id: "sample-offerless"
+  };
+  let offerDatabase: Database;
+  before(async function() {
+    offerDatabase = await getTestDatabase("has offer", undefined);
+    containerWithOffer = await getTestContainer(
+      "offerContainer",
+      undefined,
+      containerRequestWithOffer
+    );
+    containerWithoutOffer = await getTestContainer("container", undefined, containerRequest);
+    const response1 = await offerDatabase.containers.create(containerRequest);
+    const response2 = await offerDatabase.containers.create(containerRequestWithOffer);
+    container2WithOffer = response1.container;
+    container2WithoutOffer = response2.container;
+  });
+  describe("database does not have offer", function() {
+    it("has offer", async function() {
+      const offer: any = await containerWithOffer.readOffer();
+      assert.equal(offer.resource.offerVersion, "V2");
+    });
+    it("does not have offer", async function() {
+      const offer: any = await containerWithoutOffer.readOffer();
+      assert.equal(offer.resource.offerVersion, "V2");
+    });
+  });
+  describe("database has offer", function() {
+    it("container does not have offer", async function() {
+      const offer: any = await container2WithoutOffer.readOffer();
+      assert.equal(offer.resource.offerVersion, "V2");
+    });
+    it("container has offer", async function() {
+      const offer = await container2WithOffer.readOffer();
+      assert.equal(offer.resource.offerVersion, "V2");
```

Comment:
not sure yet how to verify this is the container's offer. These "v2" checks are placeholder and need to be fixed post review cc @southpolesteve

Reply:
Check that `offerResourceId` matches the container `_rid` property